include/boost/url/detail/parts_base.hpp

```diff
@@ -10,6 +10,7 @@
 #ifndef BOOST_URL_DETAIL_PARTS_BASE_HPP
 #define BOOST_URL_DETAIL_PARTS_BASE_HPP
 
+#include <boost/url/detail/config.hpp>
```

Comment:
detail files don't need to include config.hpp... you only need to include config.hpp as the first include of a public header